compiler/parser/RdrHsSyn.hs

```diff
@@ -2459,6 +2462,7 @@ mkTemplateClassInstanceSigs templateName tyVars mbKeyType =
     , ("archive",   contractId `mkFunTy` mkUpdate unitType)
     , ("toAnyTemplate", templateType `mkFunTy` anyTemplateType)
     , ("fromAnyTemplate", anyTemplateType `mkFunTy` mkParenTy (mkQualType "Optional" `mkAppTy` templateType))
+    , ("toTextTemplateId", (mkTyVar "proxy" `mkAppTy` templateType) `mkFunTy` mkQualType "Text")
```

Comment:
I wonder if we should return this wrapped into a newtype maybe even without exposing the constructor of that newtype?

Reply:
Not sure I understand what you mean, which part would you wrap into a newtype?

Reply:
Instead of returning `Text` we could return `newtype TemplateTypeRep = TemplateTypeRep Text`.

Reply:
I guess in that case it makes more sense to call it `templateTypeRep` rather than `toTextTemplateId`.

Reply:
:+1:

Reply:
or maybe  similar to how I handled `toAnyTemplate`, call the method with the proxy `_templateTypeRep` and then have a non-proxy version called `templateTypeRep`. In that case it would probably also be fine to have the underscore thing return a `Text` and do the newtype wrapping in `templateTypeRep`. Happy to leave the choice here up to you.

Reply:
Yeah, that sounds like a good plan.